src/phoenix/experimental/evals/functions/classify.py

```diff
@@ -85,13 +85,17 @@ def __init__(
         concurrency: int = 3,
         tqdm_bar_format: Optional[str] = None,
         exit_on_error: bool = True,
+        max_retries: int = 10,
         fallback_return_value: Union[Unset, Any] = _unset,
     ):
         self.generate = generation_fn
         self.fallback_return_value = fallback_return_value
         self.concurrency = concurrency
         self.tqdm_bar_format = tqdm_bar_format
         self.exit_on_error = exit_on_error
+        self.max_retries = max_retries
+        self.base_priority = 0
+        self.requeue_priority = -1
```

Comment:
Suggestion: Make `attempts_remaining` the priority and fail out when the number of attempts remaining <= 0. Only one number is needed, not three.

Reply:
I know we're keeping track of an extra variable here, but I prefer to keep "attempts remaining" and "priority" explicitly different names so it's clear how it maps to `PriorityQueue`